q2_amr/card/normalization.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+import biom
+import pandas as pd
+from q2_types.feature_data import SequenceCharacteristicsDirectoryFormat
+from rnanorm import CPM, CTF, CUF, FPKM, TMM, TPM, UQ
+
+
+def normalize(
+    table: biom.Table,
+    method: str,
+    m_trim: float = 0.3,
+    a_trim: float = 0.05,
+    gene_length: SequenceCharacteristicsDirectoryFormat = None,
+) -> pd.DataFrame:
+    # Create Dataframe with counts from biom.Table
+    counts = pd.DataFrame(
+        data=table.matrix_data.toarray(),
+        index=table.ids(axis="observation"),
+        columns=table.ids(axis="sample"),
+    ).T
+
+    if method in ["tpm", "fpkm", "uq", "cuf", "cpm"]:
+        # Raise Error if m or a-trim parameters are given with methods TPM, FPKM, UQ,
+        # CPM or CUF
+        if m_trim != 0.3 or a_trim != 0.05:
+            raise ValueError(
+                "Parameters m-trim and a-trim can only be used with methods TMM and "
+                "CTF."
+            )
```

Comment:
```
(
    method, m_trim, a_trim, _
) = TypeMap({
    (
        Str
        % Choices("tmm", "ctf"),
        Float % Range(0, 1, inclusive_start=True),
        Float % Range(0, 1, inclusive_start=True),
    ): FeatureTable[Frequency],  # Placeholder type because visualizations have no output
    (
        Str
        % Choices("tpm", "fpkm", "uq", "cuf", "cpm"),
        None,
        None,
    ): FeatureTable[Frequency],
})
```

Yes so the None types are not allowed. Alos Nonetype is not allowed. Do you have any other idea how i could solve this? 
I could set the default to None and set them to the default of 0.3 and 0.5 only if the correct methods are used. But then the default does not show up when using the --help, so that is also not ideal. 
What do you think is the best option?

Reply:
Yeah probably having the default as None... :/